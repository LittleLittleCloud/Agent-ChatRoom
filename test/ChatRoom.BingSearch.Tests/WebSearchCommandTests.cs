using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using Spectre.Console.Testing;
using Xunit;
using ChatRoom.Client.Tests;
using ChatRoom.SDK;
using Microsoft.Extensions.DependencyInjection;
using FluentAssertions;

namespace ChatRoom.WebSearch.Tests;

public class WebSearchCommandTests : IClassFixture<DefaultClientFixture>, IClassFixture<WebSearchFixture>
{
    private readonly DefaultClientFixture _fixture;
    private readonly WebSearchFixture _bingSearchFixture;
    private readonly ChatPlatformClient _client;

    public WebSearchCommandTests(DefaultClientFixture fixture, WebSearchFixture bingSearchFixture)
    {
        _fixture = fixture;
        _bingSearchFixture = bingSearchFixture;
        _client = bingSearchFixture.Command.ServiceProvider?.GetRequiredService<ChatPlatformClient>() ?? throw new InvalidOperationException("Failed to get ChatPlatformClient.");
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public async Task ItShowHelperTextTestAsync()
    {
        var app = new CommandAppTester();
        app.SetDefaultCommand<WebSearchCommand>(WebSearchCommand.Description);

        var result = await app.RunAsync("--help");

        Approvals.Verify(result.Output);
    }

    [Fact]
    public async Task ItRegisterAndUnregisterAgentsToRoomAsync()
    {
        var agents = await _client.GetRoomMembers();
        var names = agents.Select(a => a.Name).ToList();
        names.Should().Contain("bing-search");
        names.Should().Contain("google-search");
    }
}
