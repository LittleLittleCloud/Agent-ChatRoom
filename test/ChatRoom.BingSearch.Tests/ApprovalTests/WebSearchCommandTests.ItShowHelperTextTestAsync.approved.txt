DESCRIPTION:
Web search agents for chat room.
This package provides the following web search agents:
- Bing search agent
- Google search agent

To use the agent, you need to provide a configuration file.
A configuration file is a json file with the following schema:
- https://raw.githubusercontent.com/LittleLittleCloud/Agent-ChatRoom/main/schema/chatroom_bing_search_configuration_schema.json

USAGE:
    testhost.dll [OPTIONS]

OPTIONS:
    -h, --help               Prints help information
    -v, --version            Prints version information
    -c, --config <CONFIG>    Configuration file